src/time/operators.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID__SRC_TIME_OPERATORS_H
+#define SQUID__SRC_TIME_OPERATORS_H
+
+// TODO: Remove direct timercmp() calls in legacy code.
+
+inline bool
+operator <(const timeval &a, const timeval &b)
```

Comment:
The idea of moving operators into a dedicated header file sounds odd to me: Why treat operators differently than other functions?

Reply:
To restrict the amount of global namespace pollution by non-forward declarations in forward.h. These are a distinct group of operators for timeval which are only used by two callers.

If you have a better filename suggestion, I am open to ideas.

Reply:
> Alex: Why treat operators differently than other functions?
> Amos: To restrict the amount of global namespace pollution by non-forward declarations in forward.h.

These declarations do not belong to `forward.h`, so the pollution you are talking about will not exist in the finished PR and, hence, cannot be used to justify this special treatment of operators.

> Amos: If you have a better filename suggestion, I am open to ideas.

It is not a naming problem -- the group itself should not exist (and, hence, does not need a name). Please move these declarations to `time/gadgets.h` (together with all other declarations that should be there) unless you can propose a better place for them (based on their semantics rather than the fact that they are declared using an `operator` keyword).

Reply:
Undone in [d53ffe3](https://github.com/squid-cache/squid/pull/1001/commits/d53ffe3f66a8db05d3cc6327960a50594d1a80be)